### PR TITLE
52/hidden items

### PR DESCRIPTION
**Pull request by Julien-Torrent**

### CODE REVIEW

File: cypress/fixtures/apps.js [hunk 1]
```diff
@@ -17,4 +17,5 @@ export const GRAASP_APP_ITEM = {
     isPinned: false,
     showChatbox: false,
   },
+  tags: [],
```
Comment (pyphilia) [inline cypress/fixtures/apps.js:20]: I think you can avoid adding tags everywhere if you edit the mock (avoid the mock to throw).

File: cypress/fixtures/items.js [hunk 2]
```diff
@@ -123,6 +124,56 @@ export const FOLDER_WITH_PINNED_ITEMS = {
...
+      },
+      settings: {
+        isPinned: false,
+        showChatbox: false,
+      },
+      tags: [
+        {
+          tagId: false,
+        }
+      ],
```
Comment (pyphilia) [inline cypress/fixtures/items.js:171]: I might be wrong, but it doesn't look hidden? It's hidden with the taf right?

File: cypress/support/server.js [hunk 3]
```diff
@@ -125,4 +126,40 @@ export const mockDefaultDownloadFile = (items, shouldThrowError) => {
...
 
+export const mockGetItemTags = (items) => {
+  cy.intercept(
+    {
+      method: DEFAULT_GET.method,
+      url: new RegExp(`${API_HOST}/${buildGetItemTagsRoute(ID_FORMAT)}$`),
+    },
+    ({ reply, url }) => {
+      const itemId = url.slice(API_HOST.length).split('/')[2];
+      const result = items.find(({ id }) => id === itemId).tags || [];
```
Comment (pyphilia) [inline cypress/support/server.js:137]: ?.tags

File: cypress/support/server.js [hunk 4]
```diff
@@ -125,4 +126,40 @@ export const mockDefaultDownloadFile = (items, shouldThrowError) => {
...
+      method: DEFAULT_GET.method,
+      url: new RegExp(`${API_HOST}/items/tags\\?id\\=`),
+    },
+    ({ reply, url }) => {
+      const ids = url
+        .slice(API_HOST.length)
+        .split('=')
+        .splice(1)
+        .map((x) => x.replace('&id', ''));
+
```
Comment (pyphilia) [inline cypress/support/server.js:155]: Could you use `qs` to parse the ids?

File: src/components/common/Tree/CustomTreeItem.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
+/* eslint-disable react/forbid-prop-types */
```
Comment (pyphilia) [inline src/components/common/Tree/CustomTreeItem.js:1]: ?

File: src/components/common/Tree/Tree.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
+/* eslint-disable react/forbid-prop-types */
```
Comment (pyphilia) [inline src/components/common/Tree/Tree.js:1]: ?

**Review by pyphilia — CHANGES_REQUESTED** [review]
Let me know if you could edit TreeView in UI.

File: cypress/fixtures/items.js (hunk 2 above)
Comment (Julien-Torrent) [reply cypress/fixtures/items.js:170]: By default the value of REACT_APP_HIDDEN_ITEM_TAG_ID is false. So it's necessary to edit the CI env vars to add a value other than false

**Pull request merged**